### PR TITLE
Holocene: extensions to L1Block.sol

### PR DESCRIPTION
**Pull request by roberto-bayardo**
**Description**

Holocene adds [two new L1 attributes](https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/holocene/l1-attributes.md#l1-block-attributes) which need to be handled by the L1Block L1 contract.

**Tests**

Added unit and fuzz tests

**Metadata**

https://github.com/ethereum-optimism/specs/issues/340

### CODE REVIEW

**Comment by roberto-bayardo** [comment]
Hmm, decided to try vscode as my editor and it seems to have applied a standard formatting tool -- should I undo those changes or is the format it is enforcing for the files I touched preferrable?

**Comment by roberto-bayardo** [comment]
the semver-diff-check is failing, I think just because it doesn't really handle inheritance properly?

**Comment by roberto-bayardo** [comment]
Rebased and now "just snapshots-check-no-build" fails with "detected multiple artifacts for AssetReceiver". Not sure what that's from.

File: packages/contracts-bedrock/src/L2/interfaces/IL1Block.sol [hunk 1]
```diff
@@ -34,8 +34,11 @@ interface IL1Block {
     )
         external;
     function setL1BlockValuesEcotone() external;
+    function setL1BlockValuesHolocene() external;
     function timestamp() external view returns (uint64);
     function version() external pure returns (string memory);
 
     function __constructor__() external;
+    function eip1559Denominator() external view returns (uint64);
+    function eip1559Elasticity() external view returns (uint64);
```
Comment (tynes) [inline packages/contracts-bedrock/src/L2/interfaces/IL1Block.sol:41]: These should likely go above the constructor to follow style

**Comment by tynes** [comment]
Getting CI versioning checks failing

```
HEAD is now at e6cd14acb Holocene extensions to L1Block.sol
./scripts/checks/check-semver-diff.sh
✅ src/L2/L1Block.sol: version changed from 1.5.1-beta.2 to 1.5.1-beta.3
❌ Error: src/L2/L1BlockInterop.sol has changes in semver-lock.json but no version change
   Old version: +interop
   New version: +interop
error: Recipe `semver-diff-check-no-build` failed on line 158 with exit code 1

Exited with code exit status 1
```

cc @smartcontracts @mds1 

I believe this issue happened before and the solution was to add `-beta.1` to the `+interop` suffix?

File: packages/contracts-bedrock/src/L2/interfaces/IL1Block.sol (hunk 1 above)
Comment (roberto-bayardo) [reply packages/contracts-bedrock/src/L2/interfaces/IL1Block.sol:41]: done

**Comment by roberto-bayardo** [comment]
> Getting CI versioning checks failing
> 
> ```
> HEAD is now at e6cd14acb Holocene extensions to L1Block.sol
> ./scripts/checks/check-semver-diff.sh
> ✅ src/L2/L1Block.sol: version changed from 1.5.1-beta.2 to 1.5.1-beta.3
> ❌ Error: src/L2/L1BlockInterop.sol has changes in semver-lock.json but no version change
>    Old version: +interop
>    New version: +interop
> error: Recipe `semver-diff-check-no-build` failed on line 158 with exit code 1
> 
> Exited with code exit status 1
> ```
> 
> cc @smartcontracts @mds1
> 
> I believe this issue happened before and the solution was to add `-beta.1` to the `+interop` suffix?

this works

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
Perhaps a CI flake caused this to drop out of the merge queue

**Comment by tynes** [comment]
@roberto-bayardo looks like linting is blocking this

```
    packages/contracts-bedrock/src/libraries/Encoding.sol
    ❯❱ ethereum_optimism.sol-style-input-arg-fmt
          Inputs to functions must be prepended with an underscore (`_`)
          Details: https://sg.run/0jgW                                  
                                                                        
          229┆ function encodeSetL1BlockValuesHolocene(
          230┆     uint32 baseFeeScalar,
                            
  BLOCKING CODE RULES FIRED:
    ethereum_optimism.sol-style-input-arg-fmt
```

**Pull request merged**